File: tests/patterns/csharp/lambda_block_body_return.cs
class T {
  void Test() {
    // A block-bodied lambda returns only what it says; its last
    // statement is not a return.
    Action f = () => {
      // OK:
      G(t);
    };
    Func<object> k = () => {
      // ERROR:
      return G(t);
    };
  }
}

File: tests/patterns/java/lambda_block_body_return.java
class T {
  void test() {
    // A block-bodied lambda returns only what it says; its last
    // statement is not a return.
    Runnable f = () -> {
      //OK:
      g(t);
    };
    Supplier<Object> k = () -> {
      //ERROR: match
      return g(t);
    };
  }
}
